File: test-frontend.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext();
  const page = await context.newPage();
  
  const errors = [];
  page.on('console', msg => {
    if (msg.type() === 'error') {
      errors.push(`Console Error: ${msg.text()}`);
    }
  });
  page.on('pageerror', err => {
    errors.push(`Page Error: ${err.message}`);
  });
  
  const baseUrl = 'http://localhost:3000';
  const pages = [
    { path: '/', name: '首页' },
    { path: '/bid', name: '标书管理' },
    { path: '/workflow', name: '工作流' },
  ];
  
  console.log('=== 前端页面测试开始 ===\n');
  
  for (const p of pages) {
    console.log(`\n--- 测试: ${p.name} (${p.path}) ---`);
    try {
      await page.goto(`${baseUrl}${p.path}`, { waitUntil: 'networkidle', timeout: 15000 });
      console.log(`✅ 页面加载成功`);
      
      // 检查页面标题
      const title = await page.title();
      console.log(`   标题: ${title}`);
      
      // 等待一下让Vue渲染
      await page.waitForTimeout(2000);
      
      // 检查是否有错误
      const body = await page.textContent('body');
      if (body.includes('Error') || body.includes('error')) {
        console.log(`   ⚠️ 页面内容包含error`);
      }
      
    } catch (e) {
      console.log(`❌ 加载失败: ${e.message}`);
    }
  }
  
  console.log('\n\n=== 错误汇总 ===');
  if (errors.length === 0) {
    console.log('✅ 没有控制台错误');
  } else {
    errors.forEach(e => console.log(`❌ ${e}`));
  }
  
  await browser.close();
  console.log('\n=== 测试完成 ===');
})();

File: test-frontend.py
#!/usr/bin/env python3
"""前端页面Playwright测试"""

from playwright.sync_api import sync_playwright

def test_pages():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context()
        page = context.new_page()
        
        errors = []
        
        def handle_console(msg):
            if msg.type == 'error':
                errors.append(f"Console Error: {msg.text}")
        
        def handle_page_error(err):
            errors.append(f"Page Error: {err}")
        
        page.on('console', handle_console)
        page.on('pageerror', handle_page_error)
        
        base_url = 'http://localhost:3000'
        pages = [
            ('/', '首页'),
            ('/bid', '标书管理'),
            ('/workflow', '工作流'),
        ]
        
        print('=== 前端页面测试开始 ===\n')
        
        for path, name in pages:
            print(f'\n--- 测试: {name} ({path}) ---')
            try:
                response = page.goto(f'{base_url}{path}', wait_until='networkidle', timeout=15000)
                print(f'✅ 页面加载成功 (HTTP {response.status})')
                
                page.wait_for_timeout(2000)
                
                title = page.title()
                print(f'   标题: {title}')
                
                # 检查页面内容
                content = page.content()
                if 'Error' in content or 'error' in content:
                    print(f'   ⚠️ 页面包含error关键字')
                
            except Exception as e:
                print(f'❌ 加载失败: {e}')
        
        print('\n\n=== 错误汇总 ===')
        if not errors:
            print('✅ 没有控制台错误')
        else:
            for e in errors:
                print(f'❌ {e}')
        
        browser.close()
        print('\n=== 测试完成 ===')

if __name__ == '__main__':
    test_pages()